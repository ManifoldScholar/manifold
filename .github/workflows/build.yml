name: Build

on:
  push:
    branches:
      - master

jobs:
  build:
    name: Build
    runs-on: ubuntu-latest
    steps:
      - name: "checkout main"
        uses: actions/checkout@v3
        with:
          ref: ${{ github.ref_name }}

      - name: "Capture commit hash"
        run: |
          cd ${GITHUB_WORKSPACE}
          COMMIT=`git rev-parse HEAD`
          echo "commit=${COMMIT}" >> $GITHUB_ENV

      - name: "Setup Node"
        uses: ./.github/actions/setup-node

      - name: "Build client"
        run: |
          cd ${GITHUB_WORKSPACE}/client
          yarn build

      - name: "Maybe commit release changes"
        run: |
          cd ${GITHUB_WORKSPACE}
          git add --all
          git add -f client/dist
          git status
          git config --global user.email "office@castironcoding.com"
          git config --global user.name "Build Action"
          git diff --staged --quiet || git commit -m "[C] build Manifold client from ${{ env.commit }}"
          git checkout -b release-candidate

      - name: "Push release-candidate changes"
        env:
          GH_TOKEN: ${{ github.token }}
        run: |
          cd ${GITHUB_WORKSPACE}
          git push --force origin release-candidate
