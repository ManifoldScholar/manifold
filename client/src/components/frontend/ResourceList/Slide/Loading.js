import React, { Component } from "react";
import PropTypes from "prop-types";

export default class ResourceListSlideFigureLoading extends Component {
  static displayName = "ResourceList.Slide.Loading";

  static propTypes = {
    resource: PropTypes.object
  };

  componentDidMount() {
    if (!this._figure) return null;
    const parentWidth = this._figure.parentNode.offsetWidth;
    this._figure.style.width = parentWidth + "px";
  }

  render() {
    return (
      <figure>
        <div
          ref={c => {
            this._figure = c;
          }}
          className="figure-default"
          style={{ backgroundImage: "url(/static/images/resource-splash.png)" }}
        >
          <div className="resource-info">
            <i className={`manicon manicon-resource-file`} />
            <span className="resource-type">
              {"loading"}
            </span>
            <span className="resource-date">
              {"loading"}
            </span>
          </div>
        </div>
      </figure>
    );
  }
}
