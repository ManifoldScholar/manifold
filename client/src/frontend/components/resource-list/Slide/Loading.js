import React, { Component } from "react";
import PropTypes from "prop-types";
import { Icon } from "components/global/SVG";

export default class ResourceListSlideFigureLoading extends Component {
  static displayName = "ResourceList.Slide.Loading";

  static propTypes = {
    resource: PropTypes.object
  };

  componentDidMount() {
    if (!this._figure) return null;
    const parentWidth = this._figure.parentNode.offsetWidth;
    this._figure.style.width = parentWidth + "px";
  }

  render() {
    return (
      <figure>
        <div
          ref={c => {
            this._figure = c;
          }}
          className="figure-default"
          style={{ backgroundImage: "url(/static/images/resource-splash.png)" }}
        >
          <div className="resource-info">
            <i className="manicon" aria-hidden="true">
              <Icon.ResourceFile size={120} />
            </i>
            <span className="resource-type">{"loading"}</span>
            <span className="resource-date">{"loading"}</span>
          </div>
        </div>
      </figure>
    );
  }
}
