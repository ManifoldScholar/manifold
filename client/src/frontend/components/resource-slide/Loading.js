import React, { Component } from "react";
import Utility from "global/components/utility";

export default class ResourceListSlideFigureLoading extends Component {
  static displayName = "ResourceList.Slide.Loading";

  render() {
    return (
      <figure>
        <div
          className="figure-default"
          style={{ backgroundImage: "url(/static/images/resource-splash.png)" }}
        >
          <div className="resource-info">
            <Utility.IconComposer
              size={120}
              icon="resourceFile64"
              iconClass="resource-slide-figure__resource-icon"
            />
            <span className="resource-type">{"loading"}</span>
            <span className="resource-date">{"loading"}</span>
          </div>
        </div>
      </figure>
    );
  }
}
