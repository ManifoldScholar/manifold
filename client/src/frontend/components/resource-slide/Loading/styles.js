import styled from "@emotion/styled";
import Utility from "global/components/utility";
import { utilityPrimary, respond } from "theme/styles/mixins";

export const Info = styled.div`
  padding: 20px 60px;
  margin: auto;
  text-align: center;
`;

export const Icon = styled(Utility.IconComposer)`
  margin-bottom: 4px;

  ${respond(
    `
    width: 21.34vw;
    height: 21.34vw;`,
    50
  )}

  ${respond(
    `
    width: 220px;
    height: 220px;`,
    120
  )}
`;

export const Kind = styled.span`
  ${utilityPrimary}
  display: block;
  padding-bottom: 6px;
  font-size: 21px;
  font-weight: var(--font-weight-regular);
`;

export const Date = styled.span`
  ${utilityPrimary}
  font-size: 12px;
`;

export const Default = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  display: flex;
  width: 100%;
  height: 100%;
  background-position: 50% 50%;
  background-size: cover;
`;
