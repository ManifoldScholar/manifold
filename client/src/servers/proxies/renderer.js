import config from "config";
import proxy from "http-proxy-middleware";
import ch from "../../helpers/consoleHelpers";
import createStore from "../../store/createStore";
import exceptionRenderer from "../../helpers/exceptionRenderer";
import manifoldBootstrap from "bootstrap";
import CookieHelper from "helpers/cookie/Server";

const ssrRenderUrl = `http://${config.services.client.domain}:${config.services.client.sparePort}`;

export default function makeRendererProxy(requestHandler) {
  const reactServerProxy = proxy({
    target: ssrRenderUrl,
    changeOrigin: true,
    logLevel: "silent",

    onError: (err, req, res) => {
      ch.error(`Error proxying to port ${ssrRenderUrl}`);
      ch.error(`Perhaps the SSR service is in the process of reloading?`);
      ch.error(`Falling back to SSR rescue service only`);

      const store = createStore();

      const render = () => {
        try {
          requestHandler(req, res);
        } catch (error) {
          if (error.code === "MODULE_NOT_FOUND") {
            const msg =
              "Waiting for initial Webpack build to complete. Wait a few seconds " +
              "and reload.";
            ch.error(msg);
            res.setHeader("Content-Type", "text/html");
            res.end(msg);
          } else {
            ch.error(
              `Rendering fallback failed to render inn server-developmet.js`
            );
            res.setHeader("Content-Type", "text/html");
            res.end(exceptionRenderer(error));
          }
        }
      };

      const promises = [];
      const cookie = new CookieHelper(req, res);
      promises.push(manifoldBootstrap(store.getState, store.dispatch, cookie));
      Promise.all(promises).then(render, render);
    }
  });
  return reactServerProxy;
}
