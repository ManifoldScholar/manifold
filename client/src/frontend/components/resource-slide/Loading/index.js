import React from "react";
import { useTranslation } from "react-i18next";
import * as Styled from "./styles";

export default function ResourceListSlideFigureLoading() {
  const { t } = useTranslation();
  return (
    <Styled.Default
      style={{ backgroundImage: "url(/static/images/resource-splash.png)" }}
    >
      <Styled.Info>
        <Styled.Icon size={120} icon="resourceFile64" />
        <Styled.Kind>{t("common.loading")}</Styled.Kind>
        <Styled.Date>{t("common.loading")}</Styled.Date>
      </Styled.Info>
    </Styled.Default>
  );
}

ResourceListSlideFigureLoading.displayName = "ResourceList.Loading";
