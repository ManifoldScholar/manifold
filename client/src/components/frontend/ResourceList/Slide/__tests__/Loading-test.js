import React from 'react';
import renderer from 'react-test-renderer';
import Loading from '../Loading';
import build from 'test/fixtures/build';

describe("Frontend.ResourceList.Slide.Loading component", () => {

  const resource = build.entity.resource("1");

  it('renders correctly', () => {
    const component = renderer.create(
      <Loading
        resource={resource}
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});
