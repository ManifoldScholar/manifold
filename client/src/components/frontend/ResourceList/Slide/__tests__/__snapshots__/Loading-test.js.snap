// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Frontend.ResourceList.Slide.Loading component renders correctly 1`] = `
<figure>
  <div
    className="figure-default"
    style={
      Object {
        "backgroundImage": "url(/static/images/resource-splash.png)",
      }
    }
  >
    <div
      className="resource-info"
    >
      <i
        className="manicon manicon-resource-file"
      />
      <span
        className="resource-type"
      >
        loading
      </span>
      <span
        className="resource-date"
      >
        loading
      </span>
    </div>
  </div>
</figure>
`;
